ocaml/lambda/matching.ml

```diff
@@ -106,11 +106,16 @@ exception Error of Location.t * error
 let dbg = false
 
 let jkind_layout_default_to_value_and_check_not_void loc jkind =
+  let rec contains_void : Jkind.Layout.Const.t -> bool = function
+    | Any -> false
+    | Base Void -> true
+    | Base (Value | Float64 | Float32 | Word | Bits32 | Bits64) -> false
+    | Product ts -> List.exists contains_void ts
```

Comment:
What happens if `ts` is the empty list?

Reply:
The typechecker won't allow that, but I've added a sanity check.